components/org.wso2.carbon.identity.application.authenticator.organization.login/src/test/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticatorTest.java

```diff
@@ -517,6 +551,124 @@ public void testProcessWithInvalidOrgDiscoveryParam(String userEmail, String dis
         Assert.assertEquals(status, AuthenticatorFlowStatus.INCOMPLETE);
     }
 
+    @DataProvider(name = "samlIdPInitiatedB2BLoginFlowTestData")
+    public Object[][] getSAMLIdPInitiatedB2BLoginFlowTestData() {
+
+        String samlRedirectionHtmlPage = "<html>\n" +
+                "<body onload=\"javascript:document.getElementById('samlsso-response-form').submit()\">\n" +
+                "<h2>Please wait while we take you back to $acUrl</h2>\n" +
+                "<p><a href=\"javascript:document.getElementById('samlsso-response-form').submit()\">Click here</a>" +
+                " if you have been waiting for too long.</p>\n" +
+                "<form id=\"samlsso-response-form\" method=\"post\" action=\"$acUrl\">\n" +
+                "    <!--$params-->\n" +
+                "</form>\n" +
+                "</body>\n" +
+                "</html>";
+
+        return new Object[][]{
+                // When saml redirection html page is not available.
+                {false, StringUtils.EMPTY},
+                // When saml redirection html page is available.
+                {true, samlRedirectionHtmlPage}
+        };
+    }
+
+    @Test(dataProvider = "samlIdPInitiatedB2BLoginFlowTestData")
+    public void testSAMLIdPInitiatedB2BLoginFlow(boolean isSamlRedirectionHtmlPageAvailable,
+                                                 String samlRedirectionHtmlPage) throws Exception {
+
+        // Set mock context parameters for a valid organization.
+        setMockContextParamForValidOrganization();
+
+        // Set authenticator parameters.
+        authenticatorParamProperties.put(ORG_PARAMETER, orgId);
```

Comment:
```suggestion
        authenticatorParamProperties.put(ORG_PARAMETER, org);
```
Since ORG_PARAMETER value should be orgname shall we chage so?

Reply:
Addressed with https://github.com/wso2-extensions/identity-auth-organization-login/pull/49/commits/ab39e991697e6cb9ba01830b48afb8e2ac6eea67